### DISCUSSION TITLE
IDE auto-completion is missing

### DISCUSSION THREAD

**Discussion by Nesqwik** [lenra-io/lenra_cli#109] [General]
In VSCode, the Lenra JSON view have no auto-completion. 
This could be so much simplier if we could have some sort of auto-completion for the view.

**Comment by taorepoara** [comment]
It can be handle by using JSON Schema of the [components-api](https://github.com/lenra-io/components-api) to generate interface lib in the different languages.

Here are some tools to do it:

- JVM based languages (Java, Kotlin, Groovy, Scala...): https://www.jsonschema2pojo.org/
- JavaScript & TypeScript: https://www.npmjs.com/package/json-schema-to-typescript 
- Elixir: https://github.com/jschoch/jsonstruct
- Rust: https://github.com/Marwes/schemafy

And there is many others...

We could add this generation to the components-api pipeline but it can take long time ^^

**Reply by Nesqwik** [reply]
The main issue is that we have a custom meta schema. That means the generator will probably not work for this custom schema.
This need some tests :)

**Reply by taorepoara** [reply]
Do we still need the specific component type ?

**Reply by Nesqwik** [reply]
I think so. We should check to be sure tho. 
Anyway, the parsing will probably be harder if we use standard meta-schema.

**Comment by taorepoara** [comment]
Here is the tool list on jsonschema website https://json-schema.org/implementations.html#code-generation

**Comment by taorepoara** [comment]
This as been resolved by the libs for JS and TS.
Libs will be added for the other languages.
